hack/generated/controllers/generic_controller.go

```diff
@@ -498,14 +514,18 @@ func (gr *GenericReconciler) MonitorDeployment(ctx context.Context, action Recon
 		}
 	}
 
-	result := ctrl.Result{}
-	// TODO: This is going to be common... need a wrapper/helper somehow?
-	if !deployment.IsTerminalProvisioningState() {
-		result = ctrl.Result{
-			RequeueAfter: gr.RequeueDelay,
-		}
+	if deployment.IsTerminalProvisioningState() {
+		// we are done
+		return ctrl.Result{}, nil
 	}
-	return result, err
+	// need to check back; still creating resources or deleting the deployment itself
+	requeueDelay := gr.RequeueDelay
```

Comment:
Wondering if we want a helper method that just does this and returns the actual delay? It could have common logging (a few of these if's have had logs) and it saves us missing a case if we ever decide to change the logic?

Reply:
@matthchr I’m going to be updating this to remove our requeue delay in most (all?) places, given that (per discussion), controller-runtime does its own backoff if we simply specify `Requeue: true`

Reply:
Ah ok - that makes sense.

You doing that in this PR or a separate one?

Reply:
Will update this PR.